x-pack/plugins/ingest_manager/server/services/epm/agent/agent.test.ts

```diff
@@ -6,29 +6,59 @@
 
 import { createStream } from './agent';
 
-test('Test creating a stream from template', () => {
-  const streamTemplate = `
-input: log
-paths:
-{{#each paths}}
-  - {{this}}
-{{/each}}
-exclude_files: [".gz$"]
-processors:
-  - add_locale: ~
-  `;
-  const vars = {
-    paths: ['/usr/local/var/log/nginx/access.log'],
-  };
+describe('createStream', () => {
+  it('should work', () => {
+    const streamTemplate = `
+    input: log
+    paths:
+    {{#each paths}}
+      - {{this}}
+    {{/each}}
+    exclude_files: [".gz$"]
+    processors:
+      - add_locale: ~
+      `;
+    const vars = {
+      paths: { value: ['/usr/local/var/log/nginx/access.log'] },
+    };
 
-  const output = createStream(vars, streamTemplate);
+    const output = createStream(vars, streamTemplate);
+    expect(output).toEqual({
+      input: 'log',
+      paths: ['/usr/local/var/log/nginx/access.log'],
+      exclude_files: ['.gz$'],
+      processors: [{ add_locale: null }],
+    });
+  });
 
-  expect(output).toBe(`
-input: log
-paths:
-  - /usr/local/var/log/nginx/access.log
-exclude_files: [".gz$"]
-processors:
-  - add_locale: ~
-  `);
+  it('should support yaml values', () => {
```

Comment:
If we changed `createStream` to accept a map of `string` -> `any` or `unknown` (vs current `string` -> `string`) I believe we could do this test like

```typescript
test('Test nested', () => {
  const streamTemplate = `
host: {{host}}
{{#if key.patterns}}
key.patterns:{{key.patterns}}
{{/if}}
`;
  const vars = {
    host: 'http://127.0.0.1',
    key: {
      patterns: '\n  - limit: 20\n    pattern: .*',
    },
  };

  const output = createStream(vars, streamTemplate);

  expect(output).toBe(`
host: http://127.0.0.1
key.patterns:
  - limit: 20
    pattern: .*
`);
});
```

Reply:
I could retranslate to a yaml after, but it's going to be stored as a JSON and send to the agent as a JSON, so if we can avoid one other round of conversion to yaml -> to json